picopass/scenes/picopass_scene_write_key.c

```diff
@@ -17,6 +17,15 @@ NfcCommand picopass_scene_write_key_poller_callback(PicopassPollerEvent event, v
             event.data->req_write_key.key,
             picopass->write_key_context.key_to_write,
             PICOPASS_KEY_LEN);
+
+        // If there is no user dictionary, create it with the key they entered
+        // Prevent people who set all 0's from bricking their card
+        // TODO: Consider checking the elite user dict, when it exists, for the key
+        if(!iclass_elite_dict_check_presence(IclassEliteDictTypeUser)) {
+            IclassEliteDict* dict = iclass_elite_dict_alloc(IclassEliteDictTypeUser);
+            iclass_elite_dict_add_key(dict, picopass->write_key_context.key_to_write);
```

Comment:
```
(gdb) bt
#0  0x08011394 in __furi_crash_implementation () at furi/core/check.c:163
#1  0x08006636 in MemManage_Handler () at targets/f7/furi_hal/furi_hal_interrupt.c:283
#2  <signal handler called>
#3  iclass_elite_dict_add_key (dict=dict@entry=0x0, key=key@entry=0x2000ad50 "") at /Users/aku/Work/Flipper/flipperzero-good-faps/picopass/helpers/iclass_elite_dict.c:157
#4  0x2000e98c in picopass_scene_write_key_poller_callback (event=..., context=0x2000ac68) at /Users/aku/Work/Flipper/flipperzero-good-faps/picopass/scenes/picopass_scene_write_key.c:26
#5  0x2000c026 in picopass_poller_write_key_handler (instance=0x20018008) at /Users/aku/Work/Flipper/flipperzero-good-faps/picopass/protocol/picopass_poller.c:538
#6  0x2000bb86 in picopass_poller_callback (event=..., context=0x20018008) at /Users/aku/Work/Flipper/flipperzero-good-faps/picopass/protocol/picopass_poller.c:656
#7  0x08038a98 in nfc_iso14443a_poller_trx_custom_parity (instance=0x2000bb69 <picopass_poller_callback>, tx_buffer=0xa5a5a5a5, rx_buffer=0xa5a5a5a5, fwt=<optimized out>) at lib/nfc/nfc.c:446
#8  0x08038b1c in nfc_poller_trx (instance=0x0, tx_buffer=0x8, rx_buffer=0x20000284 <uxCriticalNesting>, fwt=536964632) at lib/nfc/nfc.c:480
#9  0xa5a5a5a4 in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
(gdb) f 3 
#3  iclass_elite_dict_add_key (dict=dict@entry=0x0, key=key@entry=0x2000ad50 "") at /Users/aku/Work/Flipper/flipperzero-good-faps/picopass/helpers/iclass_elite_dict.c:157
157	        if(!stream_seek(dict->stream, 0, StreamOffsetFromEnd)) break;
```

Reply:
was writing elite with all 00

Reply:
> was writing elite with all 00

Expected behavior, but I'll look into the backtrace

Reply:
I moved where the write happens, I don't think it clicked with me before that it was doing the poller interactions, which is a bad place to put it.  I moved it to the on_enter and tried it a few times locally.